app/single-instance.js

```diff
@@ -23,8 +23,15 @@ export default function () {
       // successfully connect to the running server). Just send some data to the server, which is
       // running on the first instance, so the main window can be focused there and quit this
       // instance.
-      client.write('focus first instance')
-      app.quit()
+
+      let event = 'focus first instance'
+
+      if (process.argv.length > 1) {
```

Comment:
These kind of checks should ideally be shared code with what is happening in the main process, as already even in this PR they are subtly different and have different bugs.

I think a better overall approach here would be to serialize any arguments here and pass them to the main process directly (by sending them in the `client.write`), then the main process can handle them via the same code it handles the initial launch arguments with.